foam_controlwrapper/foam_mesh.py

```diff
@@ -181,8 +184,7 @@ def __init__(self, name, BC=None, mesh=None, path=None):
             foamface.writeMesh(name)
 
             # write possible cell data to time directory
-            for cell in mesh.iter_cells():
-                self.update_cell(cell)
+            self.update_cells(list(mesh.iter_cells()))
```

Comment:
`list` could be removed:
`self.update_cells(mesh.iter_cells())`

Note though that you are using `mesh` here without testing if it is possibly `None`